src/main/java/frc/robot/controls/GameControllerDriverConfig.java

```diff
@@ -12,22 +27,68 @@
  */
 public class GameControllerDriverConfig extends BaseDriverConfig {
   private final GameController controller = new GameController(Constants.DRIVER_JOY);
+  private final DriveSub sub;
+
+  public GameControllerDriverConfig(Drivetrain drive, DriveSub sub) {
+      super(drive);
+      this.sub = sub;
+
 
-  public GameControllerDriverConfig(Drivetrain drive) {
-    super(drive);
   }
 
-  @Override
-  public void configureControls() {
+    @Override
+public void configureControls() {
     // TODO 4.1.1: Change to your auto command
-    controller.get(Button.A).onTrue(new DoNothing());
+    controller.get(Button.LB).onTrue(new AutoForwardCommand(drive, -1.0, 3.0));
+    controller.get(Button.RB).onTrue(new AutoForwardCommand(drive, 1.0, 3.0));
+    controller.get(Button.B).onTrue(new AutoSquareCommand(drive));
+    controller.get(Button.X).onTrue(new AutoTurnCommand(drive, 152.17)); //turns 180 degrees (angles for some reason broken)
+
     // TODO 4.1.3: Add Bang-Bang drive command
+    controller.get(Button.START).onTrue(new BangBangSubsystemCommand(sub, 1.0));
+    controller.get(Button.BACK).onTrue(new BangBangRotateCommand(sub, 3.0));
 
     // TODO 4.1.4: Add subsystem Bang-Bangs
+    controller.get(Button.START).onFalse(new BangBangSubsystemCommand(sub, 0));
 
     // TODO 4.2.2: Make robot spin while a button is pressed
+    controller.get(Button.A).whileTrue(new RunCommand(() -> drive.arcadeDrive(0, 1.0), drive));
+    controller.get(Button.Y).whileTrue(new RunCommand(() -> drive.arcadeDrive(0, -1.0), drive));
 
     // TODO 4.3.1: Add more triggers
+
+    // SequentialCommandGroup: Drive forward, then turn 45 degrees
+    controller.get(Button.RIGHT_JOY).onTrue(new SequentialCommandGroup(
+      new RunCommand(() -> drive.arcadeDrive(0, 0), drive),    // stop
+      new AutoForwardCommand(drive, 1.0, 3.0)                  // drive forward
```

Comment:
Your comment says this will drive forward and then turn 45 degrees, but that isn't what this command does. There are a few problems with this command:
<ol>
<li>You don't need to stop the drivetrain before making it move. You can completely remove the RunCommand.</li>
<li>RunCommands never end. This is why you use them for continuous things like driving or spinning while a button is held. If you want to do something that only runs once, like stopping the drivetrain at the end of a command, use InstantCommand. To run something for a specified time, use .withTimeout().</li>
<li>You don't have any command in this group to turn the robot 45 degrees, and you need another command if you remove the RunCommand.</li>
</ol>

Reply:
should work now